invoices/invoices.go

```diff
@@ -656,6 +656,51 @@ type HtlcAcceptDesc struct {
 	AMP *InvoiceHtlcAMPData
 }
 
+// UpdateType is an enum that describes the type of update that was applied to
+// an invoice.
+type UpdateType uint8
+
+const (
+	// UnknownUpdate indicates that the UpdateType has not been set for a
+	// given udpate. This kind of updates are not allowed.
+	UnknownUpdate UpdateType = iota
+
+	// CancelHTLCsUpdate indicates that this update cancels one or more
+	// HTLCs.
+	CancelHTLCsUpdate
+
+	// AddHTLCsUpdate indicates that this update adds one or more HTLCs.
+	AddHTLCsUpdate
+
+	// SettleHodlInvoiceUpdate indicates that this update settles one or
+	// more HTLCs from a hodl invoice.
+	SettleHodlInvoiceUpdate
```

Comment:
Haven't read the full PR, just one more state pops into my head.  Maybe there needs to be a `SettleHTLCsUpdate`? Since we have `CancelHTLCsUpdate` vs `CancelInvoiceUpdate`.